arlas-core/src/main/java/io/arlas/server/core/impl/elastic/services/ElasticCollectionReferenceService.java

```diff
@@ -103,23 +103,19 @@ public List<CollectionReference> getAllCollectionReferences(Optional<String> col
                 }
                 hits = client.search(requestBuilder.build(), CollectionReferenceParameters.class).hits().hits();
                 for (Hit<CollectionReferenceParameters> hit : hits) {
-                    try {
                         CollectionReference colRef = new CollectionReference(hit.id(), hit.source());
-                        checkIfAllowedForOrganisations(colRef, organisations);
-                        if (CollectionUtil.isCollectionPublic(colRef)) {
-                            collections.add(colRef);
-                        } else {
-                            for (String c : allowedCollections) {
-                                if (CollectionUtil.matches(c, hit.id())) {
-                                    collections.add(colRef);
-                                    break;
+                        if (checkIfAllowedForOrganisations(colRef, organisations)) {
+                            if (CollectionUtil.isCollectionPublic(colRef)) {
+                                collections.add(colRef);
+                            } else {
+                                for (String c : allowedCollections) {
+                                    if (CollectionUtil.matches(c, hit.id())) {
+                                        collections.add(colRef);
+                                        break;
+                                    }
                                 }
                             }
                         }
```

Comment:
shouldn't we throw an exception in else block ?

Reply:
No, that's exactly what we wanted to remove. We just list the collections, no need to throw an exception if a user can't see it, it's just reading

Reply:
You are right !